### ISSUE TITLE
Not able to create migration through CLI

### ISSUE THREAD

**Issue by dougg0k** [mikro-orm/mikro-orm#313]
Hello,

I am having issues with the CLI, as I am not able to create any migrations when using the CLI and when I run `npx mikro-orm debug` I get the same error, though the `dbName` is clearly defined.

It's working fine in the api, it recognizes `dbName` without any problem.

I am using the latest version `3.0.1`

```
▶ npx mikro-orm migration:create
mikro-orm migration:create

Create new migration with current schema diff

Options:
  -b, --blank          Create blank migration                          [boolean]
  -d, --dump           Dumps all queries to console                    [boolean]
  --disable-fk-checks  Do not skip foreign key checks                  [boolean]
  -p, --path           Sets path to directory where to save entities    [string]
  -v, --version        Show version number                             [boolean]
  -h, --help           Show help                                       [boolean]

Error: No database specified, please fill in `dbName` option
    at Configuration.validateOptions (/backend/node_modules/mikro-orm/dist/utils/Configuration.js:108:19)
    at new Configuration (/backend/node_modules/mikro-orm/dist/utils/Configuration.js:15:18)
    at Function.getConfiguration (/backend/node_modules/mikro-orm/dist/cli/CLIHelper.js:28:24)
    at Function.getORM (/backend/node_modules/mikro-orm/dist/cli/CLIHelper.js:34:25)
    at Function.handleMigrationCommand (/backend/node_modules/mikro-orm/dist/cli/MigrationCommandFactory.js:66:21)
```

```
▶ npx mikro-orm debug
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 3.0.1
   - node 12.14.1
   - typescript 3.7.5
 - package.json found
 - ts-node enabled
 - searched config paths:
   - /backend/src/config/db.config.ts (found)
   - /backend/dist/config/db.config.js (not found)
   - /backend/mikro-orm.config.ts (not found)
   - /backend/mikro-orm.config.js (not found)
- configuration not found (No database specified, please fill in `dbName` option)
```

`db.config.ts`
```
import { MikroORM } from "mikro-orm";
import * as entities from "../entities";

async function db(): Promise<MikroORM> {
	return await MikroORM.init({
		entities: Object.values(entities),
		dbName: "db_name",
		baseDir: __dirname + "../",
		debug: process.env.NODE_ENV === "development",
		logger: console.log.bind(console),
		tsNode: true,
		user: process.env.DATABASE_USER,
		password: process.env.DATABASE_PASSWORD,
		host: process.env.DATABASE_HOST,
		port: Number(process.env.DATABASE_PORT),
		type: "postgresql",
		forceUtcTimezone: true,
		strict: true,
		migrations: {
			tableName: "migrations",
			path: "./migrations",
			transactional: true,
			allOrNothing: true,
		},
		cache: {
			enabled: false,
		},
		discovery: {
			tsConfigPath: "../tsconfig.json",
		},
	});
}

export default db;
```
`package.json`
```
"mikro-orm": {
		"useTsNode": true,
		"configPaths": [
			"./src/config/db.config.ts",
			"./dist/config/db.config.js"
		]
},
```

**Comment by B4nan** [comment]
You are supposed to have only the configuration object in the config `db.config.ts` file, while you are exporting a promise (that would resolve in `MikroORM` instance). 

Change it to:

```typescript
import * as entities from "../entities";

const db = {
		entities: Object.values(entities),
		dbName: "db_name",
		baseDir: __dirname + "../",
		debug: process.env.NODE_ENV === "development",
		logger: console.log.bind(console), // this is default
//		tsNode: true, // you should not have this here, it will be automatically checked, the option is there basically just for running tests so you can manually enforce ts-node mode
		user: process.env.DATABASE_USER,
		password: process.env.DATABASE_PASSWORD,
		host: process.env.DATABASE_HOST,
		port: Number(process.env.DATABASE_PORT),
		type: "postgresql",
		forceUtcTimezone: true,
		strict: true,
		migrations: {
			tableName: "migrations",
			path: "./migrations",
			transactional: true, // also default value
			allOrNothing: true, // also default value
		},
		cache: {
			enabled: false,
		},
		discovery: {
			tsConfigPath: "../tsconfig.json",
		},
	});
}

export default db;
```

Or you could keep it as it is, but the default export needs to be just the configuration, so you could do:

```typescript
const options = { ... };

export async function db(): Promise<MikroORM> {
  return MikroORM.init(options);
}

export default options;
```

**Comment by dougg0k** [comment]
I actually though of that, but after a quick look at the docs I found it unclear, so I didnt try. Thanks for the help.

**Comment by B4nan** [comment]
Yeah it is not well documented, I will change the docs. 

Nowadays you can also omit the options parameter from `MikroORM.init()` and it will use the CLI config automatically.

**Comment by varunkamra** [comment]
@B4nan I am not able to create a migration either.

this is my mikro-orm.config.ts file
```
import { Options } from "@mikro-orm/core";
import { constants } from "./constants";
import { PostgreSqlDriver } from "@mikro-orm/postgresql";
import { TSMigrationGenerator } from "@mikro-orm/migrations";

const config: Options = {
  dbName: constants.DB_NAME,
  user: constants.DB_USER,
  password: constants.DB_PASSWORD,
  port: parseInt(constants.DB_PASSWORD),
  host: constants.DB_HOST,
  driver: PostgreSqlDriver, 
  migrations: {
    tableName: "mikro_orm_migrations",
    pathTs: "../migrations",
    emit: "ts",
    generator: TSMigrationGenerator,
    transactional: true,
  },
}

export default config;
```

Here is the mikro-orm config of package.json
```
"mikro-orm":{
    "useTsNode": true,
    "configPaths": [
      "./config/mikro-orm.config.ts"
    ]
  }
```

Packages I have installed:
```
    "@mikro-orm/cli": "^5.7.0",
    "@mikro-orm/core": "^5.7.0",
    "@mikro-orm/migrations": "^5.7.0",
    "@mikro-orm/postgresql": "^5.7.0",
```

Command I am trying to run
```
./node_modules/.bin/mikro-orm migration:create --blank
```

This gives me error:
```
TypeError: Cannot read properties of undefined (reading 'length')
```

Note: I do not want to use schema, I already have a init.sql file defined which creates db and db user as postgres docker entrypoint. Could you help?

**Comment by B4nan** [comment]
This issue is 3 years old, its good you checked existing ones, but you should also sort them :] 

Reported and fixed already: #4265, will ship new version later today.

**Comment by varunkamra** [comment]
Thank you!
